subprojects/docs/src/docs/userguide/running-builds/gradle_daemon.adoc

```diff
@@ -317,5 +405,63 @@ sudo sysctl -p --system
 ----
 
 Each used inotify watch takes up to 1KB of memory.
-Assuming inotify uses all the 512K watches then around 500MB will be used for watching the file system.
-If your environment is memory constraint, you may want to disable file system watching.
+Assuming inotify uses all the 512K watches then file system watching could use up to 500MB.
+In a memory constrained environment, you may want to disable file system watching.
+
+== Performance Monitoring
+
+Gradle actively monitors heap usage to detect memory leaks in the Daemon.
+When a memory leak exhausts available heap space, the Daemon:
+
+1. Finishes the currently running build
+1. Restarts before running the next build
+
+Gradle enables this monitoring by default.
+To disable this monitoring, set the `org.gradle.daemon.performance.enable-monitoring` Daemon option to
+`false`. You can do this on the command line with the following command:
+
+====
+[listing.terminal]
+----
+$ gradle <task> -Dorg.gradle.daemon.performance.enable-monitoring=false
+----
+====
+
+Or configure the property in the `gradle.properties` file in the project root or your Gradle project home:
+
+====
+.gradle.properties
+[source,properties]
+----
+org.gradle.daemon.performance.enable-monitoring=false
+----
+====
+
+[[daemon_faq]]
+== FAQ
```

Comment:
Should there be a single FAQ page with subheadings that each section could link to instead of various FAQs on various pages?